packages/state/src/VaultLeaf.ts

```diff
@@ -22,8 +22,8 @@ export class VaultLeaf extends MerkleValue {
   // https://github.com/starkware-libs/starkex-resources/blob/master/stark_ex_objects/starkware/objects/state.py#L76
   async calculateHash() {
     const key_token_hash = await pedersen(
-      PedersenHash(this.starkKey.substring(2)),
-      this.token
+      PedersenHash(this.starkKey.toString()),
+      PedersenHash(this.assetHash.toString())
```

Comment:
This suggests that even though it's a keccak hash we need to check if it starts with `0x0` in the `AssetHash` constructor, otherwise it will fail to be converted to `PedersenHash`.